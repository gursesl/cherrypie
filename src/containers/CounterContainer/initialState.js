import { fromJS } from 'immutable'

const initialState = fromJS({
  value: 0,
})

export default initialState
