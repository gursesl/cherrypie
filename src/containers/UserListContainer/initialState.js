import { fromJS } from 'immutable'
import { SELECTOR_USERS_USERS, SELECTOR_USERS_ERROR, SELECTOR_USERS_ISLOADING } from './constants'

const initialState = fromJS({
  [SELECTOR_USERS_USERS]: [
    {
      id: "95617189",
      firstName: "Elfrieda",
      lastName: "Frank",
      email: "Ada20@hotmail.com",
    },
    {
      id: "95617188",
      firstName: "Jim",
      lastName: "Smith",
      email: "jsmith@mail.com",
    },
  ],
  [SELECTOR_USERS_ERROR]: 'This is an error',
  [SELECTOR_USERS_ISLOADING]: true,
})

export default initialState
