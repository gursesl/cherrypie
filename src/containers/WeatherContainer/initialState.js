import { fromJS } from 'immutable'
import {
  SELECTOR_WEATHER_ZIP,
  SELECTOR_WEATHER_CITY,
  SELECTOR_WEATHER_RESULTS,
  SELECTOR_WEATHER_ERROR,
} from './constants'

const initialState = fromJS({
  [SELECTOR_WEATHER_ZIP]: '20165',
  [SELECTOR_WEATHER_CITY]: 'Sterling',
  [SELECTOR_WEATHER_RESULTS]: [],
  [SELECTOR_WEATHER_ERROR]: 'Load error',
})

export default initialState
