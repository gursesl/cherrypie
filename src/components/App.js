/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import CherryPieHeader from './cherryPieHeader';
import UserList from '../containers/UserListContainer';
import CounterContainer from '../containers/CounterContainer'

export default class App extends Component {
  componentDidMount() {
    // console.log('Component did mount!');
  }

  render() {
    return (
      <div>
        <nav className="ui inverted red menu">
            <h3 className="header iterm">CherryPie</h3>
            <a className="active item">Users</a>
            <a className="item">Counter</a>
            <a className="item">Tornado</a>
            <a className="item">Item 4</a>
            <a className="item">Item 5</a>
        </nav>

        <div className="ui four column doubling stackable grid">
          <div className="column">
            <h2>Services</h2>
            <p>In labore anim adipisicing aliquip exercitation dolor. Consectetur cupidatat et excepteur laboris reprehenderit. Magna proident quis dolore esse reprehenderit commodo labore anim. Excepteur anim minim cillum consequat pariatur officia ea fugiat fugiat exercitation elit occaecat. Minim id excepteur ipsum aute eiusmod laborum cillum.</p>
            <p>Do cillum velit id ea aliqua esse. Sit reprehenderit do duis aliqua sit quis pariatur amet ipsum magna reprehenderit. Anim minim sit excepteur Lorem officia cillum voluptate deserunt id.</p>
          </div>
          <div className="column">
            <h2>Prices</h2>
            <p>Veniam ipsum officia aute duis culpa deserunt sit ut. Aliquip laborum sunt aliquip commodo mollit do dolor adipisicing. Est occaecat esse dolor veniam esse ipsum in esse mollit consequat culpa. Exercitation consequat culpa nulla culpa fugiat pariatur pariatur. Mollit adipisicing in nisi Lorem ullamco esse sunt exercitation officia exercitation veniam nisi pariatur.</p>
            <p>Ex anim nisi nostrud esse culpa nostrud eu. Nulla qui magna mollit velit eiusmod est nostrud labore aliquip esse. Veniam dolor quis non non. Enim nostrud voluptate sunt dolor esse eu. Officia do commodo anim cupidatat aliqua proident. Mollit mollit est consequat officia non. Elit adipisicing enim mollit in magna.</p>
          </div>
          <div className="column">
            <h2>Team</h2>
            <p>Tempor qui do consectetur ex cillum consectetur cillum ea amet duis duis sit ex exercitation. Sunt in sint exercitation laboris duis nulla cillum anim ut ea qui nostrud. Exercitation occaecat nisi non cupidatat anim magna proident culpa. Enim proident culpa duis tempor dolor mollit. Veniam occaecat occaecat exercitation commodo laboris velit. Laborum laboris aliqua occaecat et consectetur ea. Magna cupidatat cupidatat cillum laborum aliquip magna.</p>
            <p>Sunt mollit exercitation aute sit eiusmod consequat velit. Velit consectetur nulla mollit velit incididunt ipsum. Non aliquip eu consectetur aliquip laborum exercitation ex nisi est ex ea occaecat ipsum. Amet nostrud exercitation ipsum occaecat. Qui commodo aliqua commodo aliquip dolore occaecat tempor quis adipisicing ut.</p>
          </div>
          <div className="column">
            <h2>About</h2>
            <p>Nisi culpa duis in dolor quis minim do sint irure eu dolor labore. Reprehenderit aliquip cupidatat irure labore adipisicing ad. Nulla nulla amet enim do laborum esse velit enim tempor. Pariatur minim quis esse eu incididunt dolore laboris consequat. Commodo id proident laboris nostrud reprehenderit id magna enim proident eiusmod. Aliqua sit occaecat incididunt eu adipisicing cillum esse.</p>
            <p>Deserunt aliquip consectetur ex proident quis velit ad eu mollit qui sit velit non laborum. Laborum duis veniam commodo do laboris aliquip. Laboris nisi ea mollit ad.</p>
          </div>
        </div>

        <br/><br/><br/><br/><br/><br/>

        <div className="ui attached stackable menu">
          <div className="ui container">
              <a className="item">
                <i className="home icon"></i> Home
              </a>
              <a className="item">
                <i className="grid layout icon"></i> Browse
              </a>
              <a className="item">
                <i className="mail icon"></i> Messages
              </a>
              <div className="ui simple dropdown item">
                More
                <i className="dropdown icon"></i>
                <div className="menu">
                  <a className="item"><i className="edit icon"></i> Edit Profile</a>
                  <a className="item"><i className="globe icon"></i> Choose Language</a>
                  <a className="item"><i className="settings icon"></i> Account Settings</a>
                </div>
              </div>
              <div className="right item">
                <div className="ui input"><input type="text" placeholder="Search..."/></div>
              </div>
            </div>
          </div>

          <br/><br/><br/><br/><br/><br/>



        <div className="ui grid">
          <div className="two column large screen only row">
            <div className="column">
              <div className="ui segment">
                Large Screen
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Large Screen
              </div>
            </div>
          </div>
          <div className="two column large screen only row">
            <div className="column">
              <div className="ui segment">
                Widescreen
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Widescreen
              </div>
            </div>
          </div>
          <div className="two column mobile only row">
            <div className="column">
              <div className="ui segment">
                Mobile
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Mobile
              </div>
            </div>
          </div>
          <div className="three column row">
            <div className="computer only column">
              <div className="ui segment">
                Computer
              </div>
            </div>
            <div className="tablet mobile only column">
              <div className="ui segment">
                Tablet and Mobile
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                All Sizes
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                All Sizes
              </div>
            </div>
          </div>
          <div className="four column computer only row">
            <div className="column">
              <div className="ui segment">
                Computer
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Computer
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Computer
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Computer
              </div>
            </div>
          </div>
          <div className="three column tablet only row">
            <div className="column">
              <div className="ui segment">
                Tablet
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Tablet
              </div>
            </div>
            <div className="column">
              <div className="ui segment">
                Tablet
              </div>
            </div>
          </div>
        </div>

        <div className="ui grid">
          <div className="four wide column">Incididunt consectetur labore commodo sit sunt ex. Qui incididunt nulla nulla id. Excepteur elit consequat incididunt laboris. Ea non voluptate adipisicing culpa ullamco deserunt do. Dolore laboris id culpa pariatur dolore sit nulla consequat in aliquip cillum incididunt. Dolor cupidatat in sint excepteur consequat ex nostrud consequat nostrud laboris id irure ut.</div>
          <div className="four wide column">Dolor magna id quis pariatur cupidatat ex sit culpa in anim deserunt. Laboris magna officia laboris exercitation esse tempor velit proident commodo. Lorem ullamco elit sunt adipisicing ut anim sunt eu dolor deserunt veniam ea. Tempor ea eiusmod ad culpa eu commodo nulla consectetur. Enim Lorem dolore magna et exercitation do minim non veniam nostrud incididunt sit. Minim quis eu nostrud reprehenderit sint deserunt velit nulla exercitation exercitation eiusmod laborum anim.</div>
          <div className="four wide column">Consectetur Lorem esse proident enim qui est incididunt do mollit labore occaecat eiusmod ut. Minim incididunt eiusmod elit reprehenderit tempor eu et ullamco irure Lorem officia tempor consectetur. Ipsum fugiat amet do aute minim duis.</div>
          <div className="four wide column">Eu voluptate aliqua veniam cupidatat minim elit quis enim dolor ea eu minim sint ea. Voluptate anim tempor dolor cupidatat pariatur proident pariatur officia do. Ex enim Lorem est irure aliqua incididunt adipisicing esse ullamco minim aliqua aliqua cillum. Ex labore Lorem consectetur mollit deserunt nulla anim excepteur non amet dolore pariatur irure ullamco. Aliquip tempor ut pariatur ea quis nostrud consequat officia anim dolore do.</div>
        </div>

        <h1>Hello React!</h1>
        <UserList />
        <CherryPieHeader />
        <CounterContainer />
      </div>
    );
  }
}
