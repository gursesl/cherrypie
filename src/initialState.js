import { fromJS } from 'immutable'
import { SELECTOR_COUNT, SELECTOR_COUNT_VALUE } from './containers/CounterContainer/constants'
import { SELECTOR_USERS, SELECTOR_USERS_USERS, SELECTOR_USERS_ERROR, SELECTOR_USERS_ISLOADING } from './containers/UserListContainer/constants'

export default fromJS({
  [SELECTOR_USERS]: {
    [SELECTOR_USERS_USERS]: [
      {
        id: 95617189,
        firstName: 'Elfrieda',
        lastName: 'Frank',
        email: 'Ada20@hotmail.com',
      },
      {
        id: 95617188,
        firstName: 'Jim',
        lastName: 'Smith',
        email: 'jsmith@mail.com',
      },
    ],
    [SELECTOR_USERS_ERROR]: 'This is a global error',
    [SELECTOR_USERS_ISLOADING]: true,
  },
  [SELECTOR_COUNT]: {
    [SELECTOR_COUNT_VALUE]: 10,
  },
})
