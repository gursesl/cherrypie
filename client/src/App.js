import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Route, Switch } from 'react-router-dom'
import { graphql } from 'react-apollo'
import requireAuth from './components/requireAuth'
import Home from './components/Home'
import Header from './components/Header'
import Footer from './components/Footer'
import CounterContainer from './containers/Counter'
import UserListContainer from './containers/UserList'
import WeatherContainer from './containers/Weather'
import UsersGraphQL from './components/UsersGraphQL' //eslint-disable-line
import Profile from './components/Profile'
import LoginModal from './components/LoginModal'
import LogoutModal from './components/LogoutModal'
import RegisterModal from './components/RegisterModal'

import './App.css'
import query from './graphql/queries/currentUserQuery'

// eslint-disable-next-line
class App extends Component {
  render() {
    return (
      <div className="App">
        <Header getCurrentUser={this.props.data.getCurrentUser} loading={this.props.data.loading} />
        <div className="ui">{this.props.children}</div>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/counter" component={requireAuth(CounterContainer)} />
          <Route path="/users" component={requireAuth(UserListContainer)} />
          <Route path="/gqlusers" component={requireAuth(UsersGraphQL)} />
          <Route path="/weather" component={WeatherContainer} />
          <Route path="/profile" component={Profile} />
          <Route path="/login" component={LoginModal} />
          <Route path="/logout" component={LogoutModal} />
          <Route path="/register" component={RegisterModal} />
        </Switch>
        <Footer />
      </div>
    )
  }
}

App.propTypes = {
  children: PropTypes.element,
}

// export default App
// export default withRouter(graphql(query)(App))
export default graphql(query)(App)
