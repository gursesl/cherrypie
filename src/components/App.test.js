/* eslint-disable no-unused-vars */
import React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Immutable from 'immutable'
import configureStore from 'redux-mock-store'
import sinon from 'sinon'
import {
  Button,
  Container,
  Divider,
  Grid,
  Header,
  Icon,
  Image,
  List,
  Menu,
  Segment,
  Visibility,
} from 'semantic-ui-react'
import { Router, Route, Switch } from 'react-router'
import createHistory from 'history/createBrowserHistory'
import '../setupTests'
import initialState from '../initialState'
import App, { FixedMenu } from './App'

sinon.spy(App.prototype, 'componentDidMount')

const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore(Immutable.fromJS(initialState))
const history = createHistory()
const containerDom = (
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <App />
    </ConnectedRouter>
  </Provider>
)

describe('App:index', () => {
  const container = mount(containerDom)

  beforeEach(() => {
  })

  it('should render without blowing up', () => {
    expect(container).toBeDefined();
  })


  it('should have the proper initial state', () => {
    const wrapper = shallow(containerDom)
    expect(wrapper.instance().store.getState()).toEqual(initialState)
  })

  it('should render a header', () => {
    const renderedComponent = mount(containerDom)
    expect(renderedComponent.find(Header).length).toBe(11)
  });

  it('should have a predefined number of nested React coponents', () => {
    const wrapper = shallow(React.createElement(App))
    expect(wrapper.children().filterWhere(n => typeof n.type() !== 'string').length).toEqual(8)
  })

  it('should render a child <CherryPieHeader /> component', () => {
    expect(container.find('CherryPieHeader').exists()).toBe(true)
  });

  it('should render a child <CounterContainer /> component', () => {
    expect(container.find('CounterContainer').exists()).toBe(true)
  });

  it('should render a child <UserListContainer /> component', () => {
    expect(container.find('UserListContainer').exists()).toBe(true)
  });

  it('should render a child <FixedMenu /> component', () => {
    expect(container.find('FixedMenu').exists()).toBe(false)
  });

  it('renders itself into the DOM with the correct container styles', () => {
    const wrapper = renderer.create(containerDom);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render routes', () => {
    const wrapper = mount(<Provider store={store}><ConnectedRouter history={history} /></Provider>)
    expect(wrapper.find(Router).length).toBeGreaterThanOrEqual(1)
  });

  // TODO: Implement scroll simulation and assert FixedMenu found
  it('should have FixedMenu visible if state visible parameter is set to true', () => {
    const wrapper = shallow(containerDom)
  })

  it('should have FixedMenu hidden if state visible parameter is set to false', () => {
    const wrapper = shallow(React.createElement(App))
    wrapper.instance().hideFixedMenu()
    expect(wrapper.state().visible).toBe(false)
  })

  it('calls componentDidMount', () => {
    expect(container.text()).toContain('Hello React!')
  })
})
