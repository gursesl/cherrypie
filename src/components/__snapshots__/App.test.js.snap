// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App:index renders itself into the DOM with the correct container styles 1`] = `
<div>
  <div>
    <div
      className="ui inverted vertical center aligned segment"
      style={
        Object {
          "minHeight": 700,
          "padding": "1em 0em",
        }
      }
    >
      <div
        className="ui container"
      >
        <div
          className="ui large inverted pointing secondary menu"
        >
          <a
            className="active item"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="item"
            href="/counter"
            onClick={[Function]}
          >
            Counter
          </a>
          <a
            className="item"
            href="/users"
            onClick={[Function]}
          >
            Users
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Careers
          </a>
          <div
            className="right item"
            onClick={[Function]}
          >
            <a
              className="ui inverted button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              Log in
            </a>
            <a
              className="ui inverted button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "marginLeft": "0.5em",
                }
              }
              tabIndex={undefined}
            >
              Sign Up
            </a>
          </div>
        </div>
      </div>
      <div
        className="ui text container"
      >
        <h1
          className="ui inverted header"
          style={
            Object {
              "fontSize": "4em",
              "fontWeight": "normal",
              "marginBottom": 0,
              "marginTop": "3em",
            }
          }
        >
          Imagine-a-Company
        </h1>
        <h2
          className="ui inverted header"
          style={
            Object {
              "fontSize": "1.7em",
              "fontWeight": "normal",
            }
          }
        >
          Do whatever you want when you want to.
        </h2>
        <button
          className="ui huge primary button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Get Started
          <i
            aria-hidden="true"
            className="right arrow icon"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui vertical segment"
    style={
      Object {
        "padding": "8em 0em",
      }
    }
  >
    <div
      className="ui container stackable middle aligned grid"
    >
      <div
        className="row"
      >
        <div
          className="eight wide column"
        >
          <h3
            className="ui header"
            style={
              Object {
                "fontSize": "2em",
              }
            }
          >
            We Help Companies and Companions
          </h3>
          <p
            style={
              Object {
                "fontSize": "1.33em",
              }
            }
          >
            We can give your company superpowers to do things that they never thought possible. Let us delight your customers and empower your needs... through pure data analytics.
          </p>
          <h3
            className="ui header"
            style={
              Object {
                "fontSize": "2em",
              }
            }
          >
            We Make Bananas That Can Dance
          </h3>
          <p
            style={
              Object {
                "fontSize": "1.33em",
              }
            }
          >
            Yes thats right, you thought it was the stuff of dreams, but even bananas can be bioengineered.
          </p>
        </div>
        <div
          className="right floated six wide column"
        >
          <img
            alt={undefined}
            className="ui large bordered image"
            height={undefined}
            rounded="10"
            src="https://react.semantic-ui.com/assets/images/wireframe/white-image.png"
            width={undefined}
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="center aligned column"
        >
          <button
            className="ui huge button"
            disabled={undefined}
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
          >
            Check Them Out
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui vertical segment"
    style={
      Object {
        "padding": "0em",
      }
    }
  >
    <div
      className="ui stackable internally celled equal width grid"
    >
      <div
        className="center aligned row"
      >
        <div
          className="column"
          style={
            Object {
              "paddingBottom": "5em",
              "paddingTop": "5em",
            }
          }
        >
          <h3
            className="ui header"
            style={
              Object {
                "fontSize": "2em",
              }
            }
          >
            What a Company
          </h3>
          <p
            style={
              Object {
                "fontSize": "1.33em",
              }
            }
          >
            That is what they all say about us
          </p>
        </div>
        <div
          className="column"
          style={
            Object {
              "paddingBottom": "5em",
              "paddingTop": "5em",
            }
          }
        >
          <h3
            className="ui header"
            style={
              Object {
                "fontSize": "2em",
              }
            }
          >
            I should not have gone with their competitor.
          </h3>
          <p
            style={
              Object {
                "fontSize": "1.33em",
              }
            }
          >
            <img
              alt={undefined}
              className="ui avatar image"
              height={undefined}
              src="https://react.semantic-ui.com/assets/images/avatar/large/nan.jpg"
              width={undefined}
            />
            <b>
              Nan
            </b>
             Chief Fun Officer Acme Toys
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui vertical segment"
    style={
      Object {
        "padding": "8em 0em",
      }
    }
  >
    <div
      className="ui text container"
    >
      <h3
        className="ui header"
        style={
          Object {
            "fontSize": "2em",
          }
        }
      >
        Breaking The Grid, Grabs Your Attention
      </h3>
      <p
        style={
          Object {
            "fontSize": "1.33em",
          }
        }
      >
        Instead of focusing on content creation and hard work, we have learned how to master the art of doing nothing by providing massive amounts of whitespace and generic content that can seem massive, monolithic and worth your attention.
      </p>
      <a
        className="ui large button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Read More
      </a>
      <h4
        className="ui horizontal divider header"
        style={
          Object {
            "margin": "3em 0em",
            "textTransform": "uppercase",
          }
        }
      >
        <a
          href="/"
        >
          Case Studies
        </a>
      </h4>
      <h3
        className="ui header"
        style={
          Object {
            "fontSize": "2em",
          }
        }
      >
        Did We Tell You About Our Bananas?
      </h3>
      <p
        style={
          Object {
            "fontSize": "1.33em",
          }
        }
      >
        Yes I know you probably disregarded the earlier boasts as non-sequitur filler content, but its really true. It took years of gene splicing and combinatory DNA research, but our bananas can really dance.
      </p>
      <a
        className="ui large button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        I am Still Quite Interested
      </a>
    </div>
  </div>
  <div
    className="ui inverted vertical segment"
    style={
      Object {
        "padding": "5em 0em",
      }
    }
  >
    <div
      className="ui container"
    >
      <div
        className="ui inverted stackable divided grid"
      >
        <div
          className="row"
        >
          <div
            className="three wide column"
          >
            <h4
              className="ui inverted header"
            >
              About
            </h4>
            <div
              className="ui inverted link list"
              role="list"
            >
              <a
                className="item"
                data-value={undefined}
                onClick={[Function]}
                role="listitem"
              >
                Favorite X-Men
              </a>
            </div>
          </div>
          <div
            className="seven wide column"
          >
            <h4
              className="ui inverted header"
            >
              Footer Header
            </h4>
            <p>
              Extra space for a call to action inside the footer that could help re-engage users.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <nav
    className="ui inverted red menu"
  >
    <h3
      className="header iterm"
    >
      CherryPie
    </h3>
    <a
      className="active item"
      href="/"
    >
      Users
    </a>
    <a
      className="item"
      href="/"
    >
      Counter
    </a>
    <a
      className="item"
      href="/"
    >
      Tornado
    </a>
    <a
      className="item"
      href="/"
    >
      Item 4
    </a>
    <a
      className="item"
      href="/"
    >
      Item 5
    </a>
  </nav>
  <div
    className="ui four column doubling stackable grid"
  >
    <div
      className="column"
    >
      <h2>
        Services
      </h2>
      <p>
        In labore anim adipisicing aliquip exercitation dolor. Consectetur cupidatat et excepteur laboris reprehenderit. Magna proident quis dolore esse reprehenderit commodo labore anim. Excepteur anim minim cillum
      </p>
      <p>
        Do cillum velit id ea aliqua esse. Sit reprehenderit do duis aliqua sit quis pariatur amet ipsum magna reprehenderit. Anim minim sit excepteur Lorem officia cillum voluptate deserunt id.
      </p>
    </div>
    <div
      className="column"
    >
      <h2>
        Prices
      </h2>
      <p>
        Veniam ipsum officia aute duis culpa deserunt sit ut. Aliquip laborum sunt aliquip commodo mollit do dolor adipisicing. Est occaecat
      </p>
      <p>
        Ex anim nisi nostrud esse culpa nostrud eu. Nulla qui magna mollit velit eiusmod est nostrud labore aliquip esse. Veniam dolor quis non non. Enim nostrud voluptate sunt dolor esse eu. Officia do
      </p>
    </div>
    <div
      className="column"
    >
      <h2>
        Team
      </h2>
      <p>
        Tempor qui do consectetur ex cillum consectetur cillum ea amet duis duis sit ex exercitation. Sunt in sint exercitation laboris duis nulla cillum anim ut ea qui nostrud. Exercitation occaecat nisi
      </p>
      <p>
        Sunt mollit exercitation aute sit eiusmod consequat velit. Velit consectetur nulla mollit velit incididunt ipsum. Non aliquip
      </p>
    </div>
    <div
      className="column"
    >
      <h2>
        About
      </h2>
      <p>
        Nisi culpa duis in dolor quis minim do sint irure eu dolor labore. Reprehenderit aliquip cupidatat irure labore adipisicing ad. Nulla nulla amet enim do laborum esse velit enim tempor. Pariatur minim quis esse
      </p>
      <p>
        Deserunt aliquip consectetur ex proident quis velit ad eu mollit qui sit velit non laborum. Laborum duis veniam commodo do laboris aliquip. Laboris nisi ea mollit ad.
      </p>
    </div>
  </div>
  <div
    className="ui attached stackable menu"
  >
    <div
      className="ui container"
    >
      <a
        className="item"
        href="/"
      >
        <i
          className="home icon"
        />
         Home
      </a>
      <a
        className="item"
        href="/"
      >
        <i
          className="grid layout icon"
        />
         Browse
      </a>
      <a
        className="item"
        href="/"
      >
        <i
          className="mail icon"
        />
         Messages
      </a>
      <div
        className="ui simple dropdown item"
      >
        More
        <i
          className="dropdown icon"
        />
        <div
          className="menu"
        >
          <a
            className="item"
            href="/"
          >
            <i
              className="edit icon"
            />
             Edit Profile
          </a>
          <a
            className="item"
            href="/"
          >
            <i
              className="globe icon"
            />
             Choose Language
          </a>
          <a
            className="item"
            href="/"
          >
            <i
              className="settings icon"
            />
             Account Settings
          </a>
        </div>
      </div>
      <div
        className="right item"
      >
        <div
          className="ui input"
        >
          <input
            placeholder="Search..."
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui grid"
  >
    <div
      className="two column large screen only row"
    >
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Large Screen
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Large Screen
        </div>
      </div>
    </div>
    <div
      className="two column large screen only row"
    >
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Widescreen
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Widescreen
        </div>
      </div>
    </div>
    <div
      className="two column mobile only row"
    >
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Mobile
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Mobile
        </div>
      </div>
    </div>
    <div
      className="three column row"
    >
      <div
        className="computer only column"
      >
        <div
          className="ui segment"
        >
          Computer
        </div>
      </div>
      <div
        className="tablet mobile only column"
      >
        <div
          className="ui segment"
        >
          Tablet and Mobile
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          All Sizes
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          All Sizes
        </div>
      </div>
    </div>
    <div
      className="four column computer only row"
    >
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Computer
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Computer
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Computer
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Computer
        </div>
      </div>
    </div>
    <div
      className="three column tablet only row"
    >
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Tablet
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Tablet
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="ui segment"
        >
          Tablet
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui grid"
  >
    <div
      className="four wide column"
    >
      Incididunt consectetur labore commodo sit sunt ex. Qui incididunt nulla nulla id. Excepteur elit consequat incididunt laboris. Ea non
    </div>
    <div
      className="four wide column"
    >
      Dolor magna id quis pariatur cupidatat ex sit culpa in anim deserunt. Laboris magna officia laboris exercitation esse tempor velit proident commodo. Lorem ullamco elit sunt adipisicing ut
    </div>
    <div
      className="four wide column"
    >
      Consectetur Lorem esse proident enim qui est incididunt do mollit labore occaecat eiusmod ut. Minim incididunt eiusmod elit reprehenderit tempor eu et ullamco irure Lorem officia tempor consectetur.
    </div>
    <div
      className="four wide column"
    >
      Eu voluptate aliqua veniam cupidatat minim elit quis enim dolor ea eu minim sint ea. Voluptate anim tempor dolor cupidatat pariatur proident pariatur officia do.
    </div>
  </div>
  <h1>
    Hello React!
  </h1>
  <div>
    <h1>
      User List Component
    </h1>
    <ul>
      <li>
        95617189
        : 
        Frank
        , 
        Elfrieda
         = 
        Ada20@hotmail.com
      </li>
      <li>
        95617188
        : 
        Smith
        , 
        Jim
         = 
        jsmith@mail.com
      </li>
    </ul>
    <button
      onClick={[Function]}
    >
      Fetch Users
    </button>
    <hr />
    <h4>
      Error:
      This is a global error
    </h4>
    <h4>
      Loading:
      true
    </h4>
  </div>
  <div>
    <h1>
      Header
    </h1>
    <h2
      className="ui header"
    >
      Semantic UI Header
      <div
        className="sub header"
      >
        Manage your account settings from this page
      </div>
    </h2>
  </div>
  <div>
    <h1>
      Counter
    </h1>
    <div>
      <button
        onClick={[Function]}
      >
        Increment after 1 sec
      </button>
      <button
        onClick={[Function]}
      >
        Increment
      </button>
       
      <button
        onClick={[Function]}
      >
        Decrement
      </button>
      <hr />
      <div>
        Clicked: 
        10
         times
      </div>
    </div>
  </div>
</div>
`;
